py/mpconfig.h

```diff
@@ -894,6 +894,12 @@ typedef double mp_float_t;
 #define MICROPY_PY_COLLECTIONS_ORDEREDDICT (0)
 #endif
 
+// Whether to provide the _asdict function for namedtuple
+#if !defined(MICROPY_PY_COLLECTIONS_NAMEDTUPLE__ASDICT) || !MICROPY_PY_COLLECTIONS_ORDEREDDICT
+#undef MICROPY_PY_COLLECTIONS_NAMEDTUPLE__ASDICT
```

Comment:
I'm not sure silent undefines like this is good practice.

Reply:
So, I removed undef and extra check from the push I made. mpconfig.h should be kept clean imho, and additional checks belong to .c files. And even if they aren't there, the compilation will just fail. Which is all better than making silent decisions behind user's back imho.

Reply:
I actually tried compilation with no orderreddict but with _asdict and that works so I went ahead and used the undef etc. My mistake was I didn't do a full build which indeed fails at link stage because mp_type_ordereddict is undefined.. So yes, this is much better.